### PR TITLE
 164798252 users should be able to reset their passwords 

### PR DESCRIPTION
**Pull request by anyatibrian**
### What does this PR do?

-  Enable users to be able to reset their password in case they have forgotten

### Description of tasks to be completed.

- Add tests for a password reset  functionality
- send a link to  the user's email with their password reset token
- Add pages send an email and resetting password respectively

### How should this be manually tested?

Visit the deployed application page on heroku
enter your email that you had registered with in the app,then you will recieve an email with a password reset link, follow this link that will redirect to the password reset page from where you enter and confirm your new password.

### Any background context you want to provide?

    N/A

What are the relevant pivotal tracker stories?
[#164798252](https://www.pivotaltracker.com/story/show/164798252)

### Screen shots
send password resetlink page
Screenshot 2019-05-16 at 19 45 23

Screenshot 2019-05-16 at 19 51 58

password reset page
Screenshot 2019-05-16 at 19 52 54

### CODE REVIEW

**Comment by e-ian** [comment]
@anyatibrian I see you have three commits, please squash them into one commit

**Review by patrickf949 — COMMENTED** [review]
@anyatibrian great work sofar,
Please resolve conflicts

**Review by e-ian — COMMENTED** [review]
@anyatibrian, please resolve merge conflicts

**Comment by e-ian** [comment]
@anyatibrian, check the commit message. The title should be in lower case

**Review by MuhanguziDavid — COMMENTED** [review]
Nice work @anyatibrian , the functionality works as expected. Please ensure to:
- resolve merge conflicts
- fully test the `passwordReset.js` and `passwordResetEmail.js` files.

**Comment by anyatibrian** [comment]
> Nice work @anyatibrian , the functionality works as expected. Please ensure to:
> 
>     * resolve merge conflicts
> 
>     * fully test the `passwordReset.js` and `passwordResetEmail.js` files.

am done with the project

**Review by patrickf949 — APPROVED** [review]
Tested
Feedback implemented
LGTM

**Review by MuhanguziDavid — CHANGES_REQUESTED** [review]
Nice work, however, as we talked about earlier, please add a statement to dismiss previous toasts in your action so as to avoid this

Screen Shot 2019-05-18 at 10 22 39

File: src/tests/components/passwordReset.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+describe("passwordComponent", () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = shallow(<PasswordResetComponent {...props} />);
+    instance = wrapper.instance();
+  });
+  it("should render properly", () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+  it("should ", () => {
```
Comment (MuhanguziDavid) [inline src/tests/components/passwordReset.test.js]: endeavour to use meaningful names for your tests

File: src/tests/components/passwordResetEmail.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    expect(wrapper).toMatchSnapshot();
+  });
+  it("should test password is too short ", () => {
+    wrapper.find("#email").simulate("change");
+  });
+  it("should test email errors field", () => {
+    wrapper.setProps({ errors: { email: "anyatibrian@gmail" } });
+    expect(wrapper).toMatchSnapshot();
+  });
+  it("should test email errors field", () => {
```
Comment (MuhanguziDavid) [inline src/tests/components/passwordResetEmail.test.js]: This test and the previous one have the same name!

File: src/tests/components/passwordResetEmail.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  handleOnSubmit: jest.fn()
+};
+describe("passwordResetEmailLink ", () => {
+  beforeEach(() => {
+    wrapper = shallow(<PasswordResetEmailComponent {...props} />);
+  });
+  it("should render without fail", () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+  it("should test password is too short ", () => {
```
Comment (MuhanguziDavid) [inline src/tests/components/passwordResetEmail.test.js]: As mentioned a number of times before, you should expect or assert something in your test

File: src/tests/containers/passwordResetContainer.test.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      passChange: {
+        success: { message: "your password has been reset successfully" }
+      }
+    };
+    expect(mapStateToProps(state)).toEqual({
+      success: { message: "your password has been reset successfully" }
+    });
+  });
+
+  it("handle onclick method", () => {
```
Comment (MuhanguziDavid) [inline src/tests/containers/passwordResetContainer.test.js]: please give meaningful names to your tests.

File: src/tests/containers/passwordResetContainer.test.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+      preventDefault: jest.fn()
+    };
+    instance.handleOnSubmit(e);
+    expect(props.passwordReset).toHaveBeenCalled();
+  });
+  it("should redirect on successfully registration", () => {
+    wrapper.setProps({ ...nextProps });
+    expect(props.history.push).toBeCalledWith("/login");
+  });
+  it("should redirect on successfully registration", () => {
```
Comment (MuhanguziDavid) [inline src/tests/containers/passwordResetContainer.test.js]: this has the same name as the previous test

File: src/tests/containers/passwordResetContainer.test.js (hunk 4 above)
Comment (anyatibrian) [reply src/tests/containers/passwordResetContainer.test.js]: done

File: src/tests/components/passwordReset.test.js (hunk 1 above)
Comment (anyatibrian) [reply src/tests/components/passwordReset.test.js]: done

File: src/styles/sidedrawer.scss [hunk 6]
```diff
@@ -0,0 +1,10 @@
+.sideDrawer {
+  .list-group {
+    background: yellow;
+
+    .list-group-item {
+      color: #000066;
+      border-radius: 0;
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (anyatibrian) [reply src/styles/sidedrawer.scss:10]: done

File: src/tests/containers/passwordResetContainer.test.js (hunk 5 above)
Comment (anyatibrian) [reply src/tests/containers/passwordResetContainer.test.js]: done

File: src/tests/containers/passwordResetContainer.test.js (hunk 4 above)
Comment (anyatibrian) [reply src/tests/containers/passwordResetContainer.test.js]: done

File: src/tests/components/passwordResetEmail.test.js (hunk 3 above)
Comment (anyatibrian) [reply src/tests/components/passwordResetEmail.test.js]: done

File: src/tests/components/passwordResetEmail.test.js (hunk 2 above)
Comment (anyatibrian) [reply src/tests/components/passwordResetEmail.test.js]: done

File: src/tests/components/passwordReset.test.js (hunk 1 above)
Comment (anyatibrian) [reply src/tests/components/passwordReset.test.js]: done

**Comment by anyatibrian** [comment]
> Nice work @anyatibrian , the functionality works as expected. Please ensure to:
> 
>     * resolve merge conflicts
> 
>     * fully test the `passwordReset.js` and `passwordResetEmail.js` files.

am done with the project

> Nice work, however, as we talked about earlier, please add a statement to dismiss previous toasts in your action so as to avoid this
> 
> Screen Shot 2019-05-18 at 10 22 39

done

File: src/tests/components/passwordReset.test.js (hunk 1 above)
Comment (MuhanguziDavid) [reply src/tests/components/passwordReset.test.js]: Please elaborate on how this is done, @anyatibrian

**Review by MuhanguziDavid — APPROVED** [review]
Awesome stuff @anyatibrian , LGTM

**Pull request merged**